core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesAction.java

```diff
@@ -267,7 +268,10 @@ private void replaceDataFiles(Iterable<DataFile> deletedDataFiles, Iterable<Data
       RewriteFiles rewriteFiles = table.newRewrite();
       rewriteFiles.rewriteFiles(Sets.newHashSet(deletedDataFiles), Sets.newHashSet(addedDataFiles));
       commit(rewriteFiles);
-    } catch (Exception e) {
+    }  catch (CommitStateUnknownException e) {
+      LOG.warn("Skip delete files because we can not determine whether the commit was successful or not.");
```

Comment:
I think this error message could be a little more clear. We try to write these in a form like "Cause: result". In this case, I'd recommend "Failed to check commit status: skipping data file deletion".

Reply:
Agreed, I'll fix it.

Reply:
done